hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java

```diff
@@ -150,76 +157,93 @@ public void doPostUpgradeSetup(@NonNull final Dispatch dispatch) {
         final var adminConfig = config.getConfigData(NetworkAdminConfig.class);
-        final List<AutoSysFileUpdate> autoUpdates = List.of(
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.feeSchedules(), config),
+        final List<AutoEntityUpdate<Bytes>> autoSysFileUpdates = List.of(
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) ->
+                                dispatchSynthFileUpdate(ctx, createFileID(filesConfig.feeSchedules(), config), bytes),
                         adminConfig.upgradeFeeSchedulesFile(),
                         SystemSetup::parseFeeSchedules),
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.throttleDefinitions(), config),
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) -> dispatchSynthFileUpdate(
+                                ctx, createFileID(filesConfig.throttleDefinitions(), config), bytes),
                         adminConfig.upgradeThrottlesFile(),
                         SystemSetup::parseThrottles),
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.networkProperties(), config),
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) -> dispatchSynthFileUpdate(
+                                ctx, createFileID(filesConfig.networkProperties(), config), bytes),
                         adminConfig.upgradePropertyOverridesFile(),
                         in -> parseConfig("override network properties", in)),
-                new AutoSysFileUpdate(
-                        createFileID(filesConfig.hapiPermissions(), config),
+                new AutoEntityUpdate<>(
+                        (ctx, bytes) -> dispatchSynthFileUpdate(
+                                ctx, createFileID(filesConfig.hapiPermissions(), config), bytes),
                         adminConfig.upgradePermissionOverridesFile(),
                         in -> parseConfig("override HAPI permissions", in)));
-        autoUpdates.forEach(update -> {
+        autoSysFileUpdates.forEach(update -> {
             if (update.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
                 dispatch.stack().commitFullStack();
             }
         });
+        final var autoNodeAdminKeyUpdates = new AutoEntityUpdate<Map<Long, Key>>(
+                (ctx, nodeAdminKeys) -> nodeAdminKeys.forEach(
+                        (nodeId, key) -> ctx.dispatchAdmin(b -> b.nodeUpdate(NodeUpdateTransactionBody.newBuilder()
+                                .nodeId(nodeId)
+                                .adminKey(key)
+                                .build()))),
+                adminConfig.upgradeNodeAdminKeysFile(),
+                SystemSetup::parseNodeAdminKeys);
+        if (autoNodeAdminKeyUpdates.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
+            dispatch.stack().commitFullStack();
```

Comment:
I noticed `fileService.updateAddressBookAndNodeDetailsAfterFreeze(systemContext, nodeStore);` was not moved after this in release/0.57 and here. We updated it in release/0.56 branch.

Reply:
Hi @iwsimon, wasn't that only a concern in 0.56 when we had [this](https://github.com/hashgraph/hedera-services/blob/release/0.56/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java#L195) block to initialize node metadata from the pre-DAB state?

Reply:
Yea, you are right. NodeMetadataHelper is not used any more.